app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,61 @@
+# controller for password reset
+class PasswordResetsController < ApplicationController
+  before_action :get_user, only: [:edit, :update]
+  before_action :valid_user, only: [:edit, :update]
+  before_action :check_expiration, only: [:edit, :update]
+
+  def new
+  end
+
+  def create
+    @user = User.find_by(email: params[:password_reset][:email].downcase)
+    if @user
+      @user.create_reset_digest
+      @user.send_password_reset_email
+      flash[:info] = 'Email sent with password reset instructions'
+      redirect_to root_url
+    else
+      flash.now[:danger] = 'Email address not found'
+      render 'new'
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if params[:user][:password].empty?
+      @user.errors.add(:password, "can't be empty")
+      render 'edit'
+    elsif @user.update_attributes(user_params)
+      log_in @user
+      flash[:succsess] = 'Password has been reset.'
+      redirect_to @user
+    else
+      render 'edit'
+    end
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit(:password, :password_confirmation)
+  end
+
+  def get_user
```

Comment:
__[rubocop]__
* Do not prefix reader method names with `get_`. [:ref](https://github.com/bbatsov/ruby-style-guide#accessor_mutator_method_names)

<!-- Comment by [SideCI](https://sideci.com) -->